File: src/main/kotlin/net/yttravelers/spotshare/entities/Restaurant.kt
package net.yttravelers.spotshare.entities

import net.yttravelers.spotshare.entities.codes.CuisineGenreDiv
import net.yttravelers.spotshare.entities.codes.RestaurantKindDiv
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "restaurant")
class Restaurant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var restaurantId: Int? = null

    @Column(nullable = false)
    var restaurantName: String = ""

    @ManyToOne
    @JoinColumn(name = "countryCode")
    var tourismCountry: Country? = null

    @ManyToOne
    @JoinColumns(
            JoinColumn(name="cuisineGenreDiv", referencedColumnName="division"),
            JoinColumn(name="cuisineGenreDivKey", referencedColumnName="divKey")
    )
    var cuisineGenreDiv: CuisineGenreDiv? = null

    @ManyToOne
    @JoinColumns(
            JoinColumn(name="restaurantKindDiv", referencedColumnName="division"),
            JoinColumn(name="restaurantKindDivKey", referencedColumnName="divKey")
    )
    var restaurantKindDiv: RestaurantKindDiv? = null

    var restaurantOpenTime: LocalDateTime? = null

    var restaurantCloseTime: LocalDateTime? = null

    var restaurantSummary: String = ""

    var restaurantAddress: String = ""

    var restaurantUrl: String = ""

}

File: src/main/angular/src/app/model/restaurant.ts
/**
 * 飲食店
 */
export interface IRestaurant {
  /** 飲食店ID */
  restaurantId?: number;
  /** 飲食店名称 */
  restaurantName?: string;
  /** 料理ジャンル区分 */
  cuisineGenreDiv?: number;
  /** 料理ジャンル区分名称 */
  cuisineGenreDivName?: string;
  /** 飲食店種類区分 */
  restaurantKindDiv?: number;
  /** 飲食店種類区分名称 */
  restaurantKindDivName?: string;
  /** 営業開始時間 */
  restaurantOpenTime?: string;
  /** 営業終了時間 */
  restaurantCloseTime?: string;
  /** 住所 */
  restaurantAddress?: string;
  /** URL */
  restaurantUrl?: string;
}